homeassistant/components/schluter/climate.py

```diff
@@ -0,0 +1,169 @@
+    api = hass.data[DATA_SCHLUTER_API]
+    temp_unit = hass.config.units.temperature_unit
+
+    async def async_update_data():
+        try:
+            return api.get_thermostats(session_id) or []
+        except RequestException as err:
+            raise UpdateFailed(f"Error communicating with Schluter API: {err}")
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="schluter",
+        update_method=async_update_data,
+        update_interval=SCAN_INTERVAL,
+    )
+
+    await coordinator.async_refresh()
+
+    async_add_entities(
+        SchluterThermostat(coordinator, idx, temp_unit, api, session_id)
+        for idx, ent in enumerate(coordinator.data)
+    )
+
+
+class SchluterThermostat(ClimateDevice):
+    """Representation of a Schluter thermostat."""
+
+    def __init__(self, coordinator, idx, temp_unit, api, session_id):
+        """Initialize the thermostat."""
+        self._unit = temp_unit
+        self.coordinator = coordinator
+        self.idx = idx
+        self._api = api
+        self._session_id = session_id
+        self._support_flags = SUPPORT_TARGET_TEMPERATURE
+
+    @property
+    def should_poll(self):
+        """Return if platform should poll."""
+        return False
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return self._support_flags
+
+    @property
+    def unique_id(self):
+        """Return unique ID for this device."""
+        return self.coordinator.data[self.idx].serial_number
+
+    @property
+    def device_info(self):
+        """Return information about the device."""
+        return {
+            "identifiers": {
+                (SCHLUTER_DOMAIN, self.coordinator.data[self.idx].serial_number)
+            },
+            "name": self.coordinator.data[self.idx].name,
+            "manufacturer": "Schluter",
+            "model": "Thermostat",
+            "sw_version": self.coordinator.data[self.idx].sw_version,
+        }
+
+    @property
+    def name(self):
+        """Return the name of the thermostat."""
+        return self.coordinator.data[self.idx].name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return self._unit
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self.coordinator.data[self.idx].temperature
+
+    @property
+    def hvac_mode(self):
+        """Return current operation ie. heat, idle."""
+        return (
+            HVAC_MODE_HEAT
+            if self.coordinator.data[self.idx].is_heating
+            else HVAC_MODE_OFF
+        )
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self.coordinator.data[self.idx].set_point_temp
+
+    @property
+    def hvac_modes(self):
+        """List of available operation modes."""
+        return None
```

Comment:
We have modes right? `havc_mode` returns some?

I guess this indicates a small issue. As in `hvac_mode` defines the mode the user has set (e.g., heating, cooling). If the device support just heating, `hvac_mode` should always return HVAC_MODE_HEAT. and `hvac_modes` should return a list with just heat as well.

If you want to indicate/show the current action the device is executing (e.g., idling, heating e.d.), the `hvac_action` should be used additionally

```py
    @property
    def hvac_action(self) -> Optional[str]:
        """Return the current running hvac operation if supported.

        Need to be one of CURRENT_HVAC_*.
        """
```

Reply:
I ended up doing most of what you suggested here. I switched the check on is_heating to use HVAC Action instead of mode. This is actually exactly what I wanted, because "Idle" is much more accurate than "Off". 

I experimented with returning HVAC_MODE_HEAT from hvac_modes, but with the nature of this particular in-floor heating thermostat, there really is no notion of switching modes. So when I return that list, a flame icon shows up on the entity UI that is clickable, and I don't really want that in this case. 

With this thermostat, essentially you set the temperature, and if it's set higher than the current temperature, it's heating, if not, it's idle. So I think the switch to HVAC Action makes sense and looks a lot better.

Reply:
In this case the havc mode is always heating... We should not change the backend behavior just because we don't like the frontend... that is a different problem and not a concern implementing this.

Reply:
Fair enough! I've modified the hvac_modes function to return HVAC_MODE_HEAT. This, of course, added the icon to the front end, so I added async_set_hvac_mode to avoid HA throwing a NotImplementedException when that icon is clicked.

Reply:
![Image of end result](https://imgur.com/rtiHglt)